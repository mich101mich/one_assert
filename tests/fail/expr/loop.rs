fn main() {
    one_assert::assert!(loop { break });
    one_assert::assert!(loop { break 1 });
    // one_assert::assert!(loop { }); // would be an error logically, but for compiler it just returns `!`, so it's not an error
}
