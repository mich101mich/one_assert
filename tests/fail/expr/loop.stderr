error[E0308]: mismatched types
 --> tests/fail/expr/loop.rs:2:32
  |
2 |     one_assert::assert!(loop { break });
  |                                ^^^^^ expected `bool`, found `()`
  |
help: give the `break` a value of the expected type
  |
2 |     one_assert::assert!(loop { break false });
  |                                      +++++

error[E0308]: mismatched types
 --> tests/fail/expr/loop.rs:3:38
  |
3 |     one_assert::assert!(loop { break 1 });
  |                                      ^ expected `bool`, found integer
